src/app/scheduler/components/CourseCard.tsx

```diff
@@ -0,0 +1,92 @@
+import { AddIcon, CloseIcon, InfoOutlineIcon } from '@chakra-ui/icons';
+import { Box, Text, Flex, VStack, Checkbox, IconButton, Link } from '@chakra-ui/react';
+import { useCallback } from 'react';
+
+import { Term } from '../../../shared/fetchers';
+import { useSavedCourses } from '../../../shared/hooks/useSavedCourses';
+
+export type CourseCardProps = {
+  term: Term;
+  subject: string;
+  title: string;
+  code: string;
+  colour: string;
+  pid: string;
+  selected: boolean;
+  handleChange: Function;
+};
+
+export function CourseCard({
+  term,
+  subject,
+  title,
+  code,
+  colour,
+  pid,
+  handleChange,
+  selected,
+}: CourseCardProps): JSX.Element {
+  const { addCourse, deleteCourse, contains } = useSavedCourses();
+  const courseIsSaved = contains(pid, term);
+
+  const onCourseClick = useCallback(() => {
+    if (term && pid !== '' && code !== '') {
+      if (!courseIsSaved) {
+        addCourse({ subject, code, pid, term });
+      } else {
+        deleteCourse({ subject, code, pid, term });
+      }
+    }
+  }, [term, pid, code, courseIsSaved, addCourse, subject, deleteCourse]);
+
+  const onChange = useCallback(() => {
+    handleChange();
+  }, [handleChange]);
+
+  return (
+    <Box boxShadow="md" cursor="pointer">
+      <Flex direction="row" background={!selected ? 'blackAlpha.200' : 'white'}>
+        <Flex background={colour} alignItems="center" justifyContent="center" mr="10px">
+          <Flex>
+            <Checkbox
+              backgroundColor="whiteAlpha.600"
+              colorScheme="whiteAlpha"
+              iconColor="black"
+              size="lg"
+              mx="7px"
+              isChecked={selected}
+              onChange={onChange}
```

Comment:
We want the entire card to be a clickable select (not just the checkmark). I think you can make the outermost `Box` be `as={"label"} which should make the entire thing the clickable thing?